src/transformers/models/fastspeech2_conformer/modeling_fastspeech2_conformer.py

```diff
@@ -0,0 +1,1939 @@
+            n_layers=config.pitch_predictor_layers,
+            n_chans=config.pitch_predictor_channels,
+            kernel_size=config.pitch_predictor_kernel_size,
+            dropout_rate=config.pitch_predictor_dropout,
+        )
+        # continuous pitch + FastPitch style avg
+        self.pitch_embed = FastSpeech2ConformerVarianceEmbedding(
+            out_channels=self.hidden_size,
+            kernel_size=config.pitch_embed_kernel_size,
+            padding=(config.pitch_embed_kernel_size - 1) // 2,
+            dropout_rate=config.pitch_embed_dropout,
+        )
+
+        self.energy_predictor = FastSpeech2ConformerVariancePredictor(
+            input_dim=self.hidden_size,
+            n_layers=config.energy_predictor_layers,
+            n_chans=config.energy_predictor_channels,
+            kernel_size=config.energy_predictor_kernel_size,
+            dropout_rate=config.energy_predictor_dropout,
+        )
+        # continuous energy + FastPitch style avg
+        self.energy_embed = FastSpeech2ConformerVarianceEmbedding(
+            out_channels=self.hidden_size,
+            kernel_size=config.energy_embed_kernel_size,
+            padding=(config.energy_embed_kernel_size - 1) // 2,
+            dropout_rate=config.energy_embed_dropout,
+        )
+
+        self.length_regulator = FastSpeech2ConformerLengthRegulator()
+
+        # The decoder is an encoder
+        self.decoder = FastSpeech2ConformerEncoder(
+            attention_dim=self.hidden_size,
+            attention_heads=config.num_attention_heads,
+            linear_units=config.decoder_linear_units,
+            num_blocks=config.decoder_layers,
+            input_layer=None,
+            dropout_rate=config.decoder_dropout_rate,
+            positional_dropout_rate=config.decoder_positional_dropout_rate,
+            attention_dropout_rate=config.decoder_attention_dropout_rate,
+            normalize_before=config.decoder_normalize_before,
+            concat_after=config.decoder_concat_after,
+            positionwise_conv_kernel_size=config.positionwise_conv_kernel_size,
+            macaron_style=config.use_macaron_style_in_conformer,
+            use_cnn_module=config.use_cnn_in_conformer,
+            cnn_module_kernel=config.decoder_kernel_size,
+            utt_embed=config.utt_embed_dim,
+        )
+
+        self.speech_decoder_postnet = FastSpeech2ConformerSpeechDecoderPostnet(config)
+
+        self.criterion = FastSpeech2ConformerLoss(
+            use_masking=config.use_masking, use_weighted_masking=config.use_weighted_masking
+        )
+
+        self.post_init()
+
+    @replace_return_docstrings(output_type=FastSpeech2ConformerModelOutput, config_class=_CONFIG_FOR_DOC)
+    def forward(
+        self,
+        input_ids: torch.LongTensor,
+        input_lengths: Optional[torch.LongTensor] = None,
+        target_spectrograms: Optional[torch.FloatTensor] = None,
+        spectrogram_lengths: Optional[torch.LongTensor] = None,
```

Comment:
Updated the model and tokenizer to use `attention_mask` instead.

Re `labels`, FastSpeech2 is somewhat unique in that it takes in many labels (spectrograms, pitch, energy, and duration) for training. I'm not sure exactly what this means for compatibility with `Trainer` since I haven't had time to do a deeper dive, but for now I've changed the "targets" to include `_labels` in their name, left the training test as skipped, and planning to look into it more when I do the demo notebook.

Reply:
AFAIK, Trainer will use any input that has `labels` in its name as the targets, so to have multiple targets you'd indeed have multiple arguments named `something_labels`.